lightning-invoice/src/payment.rs

```diff
@@ -562,22 +630,26 @@ where
 				} else if *rejected_by_dest {
 					log_trace!(self.logger, "Payment {} rejected by destination; not retrying", log_bytes!(payment_hash.0));
 					self.payer.abandon_payment(payment_id.unwrap());
+					self.remove_path_inflight_htlcs(path);
```

Comment:
What's the reasoning behind not adding `remove_path_inflight_htlcs` in the branch above (`payment_id.is_none()`) ?

Reply:
no reason, I missed it. good catch 😅